### PR TITLE
Optimize construction of headers.

### PR DESCRIPTION
**Pull request by marshall-lee**
...And remove some cryptic regexp stuff from headers construction.
New algorithm is easier to read and also faster.

### CODE REVIEW

**Comment by dblock** [comment]
I think this is a good opportunity to introduce a basic set of tests for this class, doesn't look like we have any.

**Comment by marshall-lee** [comment]
@dblock https://github.com/ruby-grape/grape/blob/5154b36ea924e7e17805d905c8e8e072f1cb1299/spec/grape/endpoint_spec.rb#L123

**Comment by dblock** [comment]
I mean a standalone test for this class alone, not in context of a request.

**Comment by marshall-lee** [comment]
@dblock ok, i'll write some.

**Comment by marshall-lee** [comment]
@dblock 

I added some basic tests, please look at it!

> is not case-sensitive

I don't think it's necessay. Remember #1055?

**Comment by dblock** [comment]
This is great. Since we have lib/grape/http/request.rb lets move the spec into spec/grape/http/request_spec.rb?

**Comment by marshall-lee** [comment]
@dblock 

I suggest to move `lib/grape/http/request.rb` to `lib/grape/request.rb` because class is named `Grape::Request` :)

**Comment by dblock** [comment]
I am with you. Do it.

**Comment by marshall-lee** [comment]
Done! :)

File: lib/grape.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ module Grape
 
     autoload :Cookies
     autoload :Validations
-    autoload :Request, 'grape/http/request'
+    autoload :Request
     autoload :Env, 'grape/util/env'
```
Comment (dblock) [inline lib/grape.rb]: You have your work cut out for you @marshall-lee right here :)

**Comment by dblock** [comment]
Merged, thanks!

**Pull request merged**